lib/resty/upstream/healthcheck.lua

```diff
@@ -225,50 +225,65 @@ local function check_peer(ctx, id, peer, is_backup)
             errlog("failed to connect to ", name, ": ", err)
         end
         peer_fail(ctx, is_backup, id, peer)
-    else
-        local bytes, err = sock:send(req)
-        if not bytes then
+        return
+    end
+
+    if ctx.type == "https" then
+        local session, err = sock:sslhandshake(nil, name, ctx.ssl_verify)
```

Comment:
I wonder if we could expose an option to reuse SSL sessions (directly saved in the healthcheck object, for example). This can reduce the healthcheck overhead considerably since a full SSL handshake is very expensive in CPU time.